kernel/sched.c

```diff
@@ -1020,6 +1020,8 @@ s32_t z_impl_k_sleep(int ms)
 {
 	s32_t ticks;
 
+	__ASSERT(!arch_is_in_isr(), "");
```

Comment:
`__ASSERT_NO_MSG` would be a better choice.

Reply:
I just replicated existing practice in this file.